api/src/main/java/org/pmiops/workbench/workspaces/VwbWorkspaceServiceImpl.java

```diff
@@ -0,0 +1,229 @@
+      WsmClient wsmClient,
+      WorkspaceMapper workspaceMapper,
+      FirecloudMapper firecloudMapper,
+      WorkspaceDao workspaceDao,
+      InitialCreditsService expirationService,
+      WorkspaceAuthService workspaceAuthService) {
+    this.wsmClient = wsmClient;
+    this.workspaceMapper = workspaceMapper;
+    this.firecloudMapper = firecloudMapper;
+    this.workspaceDao = workspaceDao;
+    this.expirationService = expirationService;
+    this.workspaceAuthService = workspaceAuthService;
+  }
+
+  @Override
+  public WorkspaceResponse getWorkspace(String workspaceNamespace, String workspaceId) {
+
+    DbWorkspace dbWorkspace = workspaceDao.getRequired(workspaceNamespace, workspaceId);
+    workspaceAuthService.validateWorkspaceTierAccess(dbWorkspace);
+
+    RawlsWorkspaceDetails fcWorkspace;
+    WorkspaceResponse workspaceResponse = new WorkspaceResponse();
+
+    WorkspaceDescription workspaceDescription =
+        wsmClient.getWorkspaceAsService(dbWorkspace.getWorkspaceNamespace());
+    fcWorkspace = workspaceMapper.toWorkspaceDetails(workspaceDescription);
+
+    workspaceResponse.setAccessLevel(
+        firecloudMapper.fcToApiWorkspaceAccessLevel(
+            firecloudMapper.fromIamRole(workspaceDescription.getHighestRole())));
+    Workspace workspace =
+        workspaceMapper.toApiWorkspace(dbWorkspace, fcWorkspace, expirationService);
+    workspaceResponse.setWorkspace(workspace);
+
+    return workspaceResponse;
+  }
+
+  @Override
+  public boolean notebookTransferComplete(String workspaceNamespace, String workspaceId) {
+    return false;
+  }
+
+  @Override
+  public List<WorkspaceResponse> getWorkspaces() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public List<WorkspaceResponse> getFeaturedWorkspaces() {
+    logger.warn("getFeaturedWorkspaces not implemented in VWB");
+    return Collections.emptyList();
+  }
+
+  @Override
+  public String getPublishedWorkspacesGroupEmail() {
+    logger.warn("getPublishedWorkspacesGroupEmail not implemented in VWB");
+    return null;
+  }
+
+  @Override
+  public void deleteWorkspace(DbWorkspace dbWorkspace) {
+    logger.warn("deleteWorkspace not implemented for VWB");
+  }
+
+  @Override
+  public void updateWorkspaceBillingAccount(DbWorkspace workspace, String newBillingAccountName) {
+    logger.warn("updateWorkspaceBillingAccount not implemented in VWB");
+  }
+
+  @Override
+  public DbWorkspace saveAndCloneCohortsConceptSetsAndDataSets(DbWorkspace from, DbWorkspace to) {
+    logger.warn("saveAndCloneCohortsConceptSetsAndDataSets not implemented in VWB");
+    return null;
+  }
+
+  @Override
+  public List<UserRole> getFirecloudUserRoles(String workspaceNamespace, String firecloudName) {
+    logger.warn("getFirecloudUserRoles not implemented in VWB");
+    return null;
+  }
+
+  @Override
+  public List<DbUserRecentWorkspace> getRecentWorkspaces() {
+    logger.warn("getRecentWorkspaces not implemented in VWB");
+    return null;
```

Comment:
https://github.com/all-of-us/workbench/blob/c4f0957478f2dbf3c2ba15a769344a30134a4d8a/api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java#L366 
this can be the same?

Reply:
it won't make a difference because this method is not called from this service. Also we still don't show a list of workspaces for VWB anyway.